File: core/src/main/java/y2k/joyreactor/Comment.java
package y2k.joyreactor;

import java.io.Serializable;

/**
 * Created by y2k on 28/09/15.
 */
public class Comment implements Serializable {

    public long postId;

    public String text;
    public String userImage;

    public long id;
    public long parentId;

    public int replies;
    public float rating;

    public UserImage getUserImage() {
        return userImage == null ? new UserImage() : new UserImage(userImage);
    }

    public Image getAttachment() {
        return attachment;
    }

    public void setAttachment(String url, int width, int height) {
        attachment = new Image(url, width, height);
    }

    private Image attachment;
}

File: core/src/main/kotlin/y2k/joyreactor/model/Comment.kt
package y2k.joyreactor.model

import com.j256.ormlite.field.DataType
import com.j256.ormlite.field.DatabaseField
import java.io.Serializable

/**
 * Created by y2k on 28/09/15.
 */
data class Comment(
    @DatabaseField val text: String = "",
    @DatabaseField val userImage: String? = null,
    @DatabaseField val parentId: Long = 0,
    @DatabaseField val rating: Float = 0f,
    @DatabaseField val postId: Long = 0,
    @DatabaseField val level: Int = 0,
    @DatabaseField var replies: Int = 0, // TODO: сделать immutable
    @DatabaseField(dataType = DataType.SERIALIZABLE) val attachment: Image? = null,
    @DatabaseField(id = true) val id: Long = 0) : Serializable {

    val userImageObject: UserImage
        get() = if (userImage == null) UserImage() else UserImage(userImage)
}